azure/services/networkinterfaces/networkinterfaces.go

```diff
@@ -19,158 +19,86 @@ package networkinterfaces
 import (
 	"context"
 
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-02-01/network"
-	"github.com/Azure/go-autorest/autorest/to"
-	"github.com/pkg/errors"
-
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/async"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/resourceskus"
+	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
+const serviceName = "interfaces"
+
 // NICScope defines the scope interface for a network interfaces service.
 type NICScope interface {
 	azure.ClusterDescriber
-	NICSpecs() []azure.NICSpec
+	azure.AsyncStatusUpdater
+	NICSpecs() []azure.ResourceSpecGetter
 }
 
 // Service provides operations on Azure resources.
 type Service struct {
 	Scope NICScope
 	Client
```

Comment:
is `Client` still being used anywhere?

Reply:
Yep, it looks like it's being used in the VM service.

Reply:
not the Client interface, the Client in the Service struct. I don't think you're using this Client anywhere in the networkinterfaces service.